modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/java/com/liferay/portal/workflow/metrics/internal/model/listener/UserModelListener.java

```diff
@@ -0,0 +1,130 @@
+import com.liferay.portal.search.query.BooleanQuery;
+import com.liferay.portal.search.query.Queries;
+import com.liferay.portal.search.query.TermsQuery;
+import com.liferay.portal.search.script.ScriptBuilder;
+import com.liferay.portal.search.script.ScriptType;
+import com.liferay.portal.search.script.Scripts;
+import com.liferay.portal.workflow.metrics.internal.petra.executor.WorkflowMetricsPortalExecutor;
+import com.liferay.portal.workflow.metrics.internal.search.index.WorkflowMetricsIndex;
+
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.component.annotations.ReferencePolicyOption;
+
+/**
+ * @author Feliphe Marinho
+ */
+@Component(immediate = true, service = ModelListener.class)
+public class UserModelListener extends BaseModelListener<User> {
+
+	@Override
+	public void onAfterUpdate(User user) throws ModelListenerException {
+		if (_nameModified) {
+			ScriptBuilder builder = _scripts.builder();
+
+			builder.idOrCode(
+				StringUtil.read(
+					getClass(),
+					"dependencies/workflow-metrics-update-task-assignee-" +
+						"script.painless")
+			).language(
+				"painless"
+			).putParameter(
+				"userId", user.getUserId()
+			).putParameter(
+				"userName", user.getFullName()
+			).scriptType(
+				ScriptType.INLINE
+			);
+
+			BooleanQuery nestedBooleanQuery = _queries.booleanQuery();
+
+			TermsQuery termsQuery = _queries.terms("tasks.assigneeIds");
+
+			termsQuery.addValues(
+				Stream.of(
+					user.getUserId()
+				).map(
+					String::valueOf
+				).toArray(
+					Object[]::new
+				));
+
+			nestedBooleanQuery.addMustQueryClauses(termsQuery);
+
+			nestedBooleanQuery.addMustQueryClauses(
+				_queries.term("tasks.assigneeType", User.class.getName()));
+
+			_workflowMetricsPortalExecutor.execute(
+				() -> searchEngineAdapter.execute(
+					new UpdateByQueryDocumentRequest(
+						_queries.nested("tasks", nestedBooleanQuery),
+						builder.build(),
+						_instanceWorkflowMetricsIndex.getIndexName(
+							user.getCompanyId()))));
+		}
+	}
+
+	@Override
+	public void onBeforeUpdate(User user) throws ModelListenerException {
+		User currentUser = _userLocalService.fetchUserById(user.getUserId());
+
+		_nameModified = !Objects.equals(
+			currentUser.getFullName(), user.getFullName());
+	}
+
+	@Reference(
+		cardinality = ReferenceCardinality.OPTIONAL,
+		policy = ReferencePolicy.DYNAMIC,
+		policyOption = ReferencePolicyOption.GREEDY,
+		target = "(search.engine.impl=Elasticsearch)"
+	)
+	protected volatile SearchEngineAdapter searchEngineAdapter;
+
+	@Reference(target = "(workflow.metrics.index.entity.name=instance)")
+	private WorkflowMetricsIndex _instanceWorkflowMetricsIndex;
+
+	private boolean _nameModified;
```

Comment:
Hey @MarinhoFeliphe do not use an stateful approach here. It'll be error prone when this ModelListener receives concurrent accesses(more than one thread at once). Please use a stateless approach(just move the code to the same method).


```java
@Override
public void onBeforeUpdate(User user) throws ModelListenerException {
  User currentUser = _userLocalService.fetchUserById(user.getUserId());
  
  TransactionCommitCallbackUtil.registerCallback(
    () -> {
    
      if(!Objects.equals(currentUser.getFullName(), user.getFullName())) {
          //execute update here
       }
    });
  }
```

Thx